fc_util/src/arg_parser.rs

```diff
@@ -0,0 +1,586 @@
+            default_value: None,
+            optional: false,
+            help: None,
+            user_value: None,
+        }
+    }
+
+    pub fn required(mut self, required: bool) -> ArgInfo<'a> {
+        self.required = required;
+        self
+    }
+
+    pub fn requires(mut self, requires: &'a str) -> ArgInfo<'a> {
+        self.requires = Some(requires);
+        self
+    }
+
+    pub fn takes_value(mut self, takes_value: bool) -> ArgInfo<'a> {
+        self.takes_value = takes_value;
+        self
+    }
+
+    pub fn default_value(mut self, default_value: &'a str) -> ArgInfo<'a> {
+        self.default_value = Some(default_value);
+        self
+    }
+
+    pub fn optional(mut self, optional: bool) -> ArgInfo<'a> {
+        self.optional = optional;
+        self
+    }
+
+    pub fn help(mut self, help: &'a str) -> ArgInfo<'a> {
+        self.help = Some(help);
+        self
+    }
+
+    pub fn set_user_value(mut self, value: String) -> ArgInfo<'a> {
+        self.user_value = Some(value);
+        self
+    }
+
+    pub fn value(&self) -> Option<String> {
+        if let Some(user_value) = &self.user_value {
+            return Some(user_value.clone());
+        } else if let Some(default_value) = self.default_value {
+            return Some(default_value.to_string());
+        }
+        None
+    }
+
+    pub fn format_help(&self) -> String {
+        let mut help_builder = vec![];
+
+        help_builder.push(format!("--{}", self.name));
+
+        if self.takes_value {
+            help_builder.push(format!(" <{}>", self.name));
+        }
+
+        if let Some(help) = self.help {
+            help_builder.push(format!(": {}", help));
+        }
+        help_builder.concat()
+    }
+}
+
+impl<'a> PartialEq for ArgInfo<'a> {
+    fn eq(&self, other: &Self) -> bool {
+        self.name == other.name
+    }
+}
+
+impl<'a> Hash for ArgInfo<'a> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.name.hash(state);
+    }
+}
+
+impl<'a> Borrow<str> for ArgInfo<'a> {
+    fn borrow(&self) -> &str {
+        self.name
+    }
+}
+
+#[derive(Debug, Default, PartialEq)]
+pub struct ArgParser<'a> {
+    args_matches: HashSet<ArgInfo<'a>>,
+    extra_args: Vec<String>,
+    help_header: String,
+}
+
+impl<'a> ArgParser<'a> {
+    pub fn arg(mut self, arg_info: ArgInfo<'a>) -> Self {
+        self.args_matches.insert(arg_info);
+        self
+    }
+
+    pub fn help(mut self, header: String) -> Self {
+        self.help_header = header;
+        self
+    }
+
+    fn print_help(self) {
+        println!("{}\n", self.help_header);
+        for arg in &self.args_matches {
+            println!("{}", arg.format_help());
+        }
+    }
+
+    pub fn value(&self, arg_name: &'static str) -> Option<String> {
+        self.args_matches.get(arg_name).unwrap().user_value.clone()
+    }
+
+    pub fn extra_args(self) -> Vec<String> {
+        self.extra_args
+    }
+
+    pub fn split(args: &[String]) -> (&[String], &[String]) {
```

Comment:
nit: I would rename this method in order to be more suggestive. `split_args` maybe. Also a comment about what this method does would be very helpful.